src/tests/ftest/util/ior_utils.py

```diff
@@ -20,167 +20,317 @@
+        """Get the total bytes expected to be written by ior.
+
+        Args:
+            processes (int): number of processes running the ior command
+
+        Returns:
+            int: total number of bytes written
+
+        """
+        power = {"k": 1, "m": 2, "g": 3, "t": 4}
+        total = processes
+        for name in ("block_size", "segment_count"):
+            item = getattr(self, name).value
+            if item:
+                sub_item = re.split(r"([^\d])", str(item))
+                if len(sub_item) > 0:
+                    total *= int(sub_item[0])
+                    if len(sub_item) > 1:
+                        key = sub_item[1].lower()
+                        if key in power:
+                            total *= 1024**power[key]
+                        else:
+                            raise IorFailed(
+                                "Error obtaining the IOR aggregate total from "
+                                "the {} - bad key: value: {}, split: {}, "
+                                "key: {}".format(name, item, sub_item, key))
+                else:
+                    raise IorFailed(
+                        "Error obtaining the IOR aggregate total from the {}: "
+                        "value: {}, split: {}".format(name, item, sub_item))
+        return total
+
+    def get_launch_command(self, basepath, processes, hostfile, runpath=None):
+        """Get the process launch command used to run IOR.
+
+        Args:
+            basepath (str): DAOS base path
+            processes (int): number of host processes
+            hostfile (str): file defining host names and slots
+            runpath (str, optional): Optional path to the mpirun/oretrun
+                command. Defaults to None.
+
+        Raises:
+            IorFailed: if an error occured building the IOR command
+
+        Returns:
+            str: ior launch command
+
+        """
+        with open(os.path.join(basepath, ".build_vars.json")) as afile:
+            build_paths = json.load(afile)
+        attach_info_path = os.path.join(basepath, "install/tmp")
+
+        if self.api.value == "MPIIO":
+            env = {
+                "CRT_ATTACH_INFO_PATH": attach_info_path,
+                "DAOS_POOL": str(self.daos_pool),
+                "MPI_LIB": "",
+                "DAOS_SVCL": str(self.daos_svcl),
+                "DAOS_SINGLETON_CLI": 1,
+                "FI_PSM2_DISCONNECT": 1,
+            }
+            export_cmd = [
+                "export {}={}".format(key, val) for key, val in env.items()]
+            mpirun_cmd = [
+                os.path.join(
+                    runpath if runpath else build_paths["OMPI_PREFIX"],
+                    "bin/mpirun"),
+                "-np {}".format(processes),
+                "--hostfile {}".format(hostfile),
+            ]
+            command = " ".join(mpirun_cmd + [self.__str__()])
+            command = "; ".join(export_cmd + command)
```

Comment:
this should be command = "; ".join(export_cmd + [command])
or else it will return TyepError